src/Sulu/Component/HttpCache/Handler/TagsHandler.php

```diff
@@ -48,51 +43,30 @@ class TagsHandler implements
 
     /**
      * @param ProxyClientInterface $proxyClient
-     * @param ContentTypeManagerInterface $contentTypeManager
+     * @param ReferenceStorePoolInterface $referenceStore
      */
-    public function __construct(
-        ProxyClientInterface $proxyClient,
-        ContentTypeManagerInterface $contentTypeManager
-    ) {
+    public function __construct(ProxyClientInterface $proxyClient, ReferenceStorePoolInterface $referenceStore)
+    {
         $this->proxyClient = $proxyClient;
-        $this->contentTypeManager = $contentTypeManager;
+        $this->referenceStore = $referenceStore;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function invalidateStructure(StructureInterface $structure)
     {
         $this->structuresToInvalidate[$structure->getUuid()] = $structure;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function updateResponse(Response $response, StructureInterface $structure)
     {
-        $tags = [
-            $this->getBanKey($structure->getUuid()),
-        ];
-
-        foreach ($structure->getProperties(true) as $property) {
-            foreach ($this->getReferencedUuids($property) as $uuid) {
-                $tags[] = $this->getBanKey($uuid);
-            }
-        }
+        $tags = array_merge([$structure->getUuid()], $this->referenceStore->getReferences());
 
-        $response->headers->set(
-            self::TAGS_HEADER,
-            implode(',', $tags)
-        );
-    }
-
-    private function getReferencedUuids(PropertyInterface $property)
-    {
-        $contentTypeName = $property->getContentTypeName();
-        $contentType = $this->contentTypeManager->get($contentTypeName);
-        $referencedUuids = $contentType->getReferencedUuids($property);
-
-        return $referencedUuids;
-    }
-
-    private function getBanKey($uuid)
-    {
-        return 'structure-' . $uuid;
+        $response->headers->set(self::TAGS_HEADER, implode(',', $tags));
```

Comment:
What I don't like about the `__toString` method, is that it makes these variables look like if they are a string, although they aren't... I also like what @wachterjohannes has already suggested: Moving the string concatenation of type and identifier in here, since we might need different string representation in other places. The question I ask myself is if we shouldn't do it like this then, instead of using a separate `Reference` class:

```php
$tags = [];
foreach ($this->referencePoolStore as $alias => $referenceStore) {
    foreach ($referenceStore->getReferences() => $reference) {
        $tags[] = $alias . '-' . $reference;
    }
}
```